go-controller/pkg/ovn/secondary_localnet_network_controller.go

```diff
@@ -353,11 +402,77 @@ func (oc *SecondaryLocalnetNetworkController) newRetryFramework(
 }
 
 func (oc *SecondaryLocalnetNetworkController) localnetPortNetworkNameOptions() map[string]string {
-	localnetLSPOptions := map[string]string{
-		"network_name": oc.GetNetworkName(),
+	return map[string]string{
+		"network_name": oc.findNetworkName(),
+	}
+}
+
+func (oc *SecondaryLocalnetNetworkController) checkBridgeMapping(obj interface{}) error {
+	if !config.OVNKubernetesFeature.EnableInterconnect {
+		return nil
+	}
+
+	pod, isPod := obj.(*corev1.Pod)
+	if !isPod {
+		return fmt.Errorf("could not cast %T object to *knet.Pod", obj)
```

Comment:
ovn-k error are most of the time prefixed with "failed"
```suggestion
		return fmt.Errorf("failed casting %T object to *knet.Pod", obj)
```